core/src/main/java/hivemall/regression/RegressionBaseUDTF.java

```diff
@@ -228,14 +245,61 @@ protected PredictionResult calcScoreAndVariance(@Nonnull final FeatureValue[] fe
 
     protected void update(@Nonnull final FeatureValue[] features, float target, float predicted) {
         float d = computeUpdate(target, predicted);
-        update(features, d);
+
+        if (this.is_mini_batch) {
+            batchUpdate(features, d);
+        } else {
+            onlineUpdate(features, d);
+        }
     }
 
     protected float computeUpdate(float target, float predicted) {
         throw new IllegalStateException();
     }
 
-    protected void update(@Nonnull final FeatureValue[] features, float coeff) {
+    protected IWeightValue getNewWeight(IWeightValue old_w, float delta) {
+        throw new IllegalStateException();
+    }
+    /**
+     * Accumulates the delta calculated from each samples
+     * @param features
+     * @param coeff
+     */
+    protected void accumulateDelta(@Nonnull final FeatureValue[] features, float coeff) {
+        for (int i = 0; i < features.length; i++) {
+            if (features[i] == null) {
+                continue;
+            }
+            final Object x = features[i].getFeature();
+            final float xi = features[i].getValue();
+            float delta = xi * coeff;
+            if (accDelta[i] == null) {
+                accDelta[i] = new FeatureValue(x, delta);
+            } else {
+                accDelta[i].setValue(accDelta[i].getValue() + delta);
```

Comment:
I doubt that 32 bits `float` is enough for accumulating hundreds of training instances.
Changes value of `FeatureValue` to `double`, possibly?